### PR TITLE
OE-5442 Leaflet firm and subspecialty admin list combined together v1

### PR DESCRIPTION
**Pull request by vetusko**

### CODE REVIEW

File: protected/controllers/oeadmin/LeafletSubspecialtyFirmController.php [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+				&& isset($search['filterid']['firm_id']['value']))
+			|| ($subspecialtyId == $session['lastSubspecialtyId'] && $search['filterid']['firm_id']['value'] == "")
+		) {
+			$session['lastSubspecialtyId'] = "";
+			$this->redirect('/oeadmin/LeafletSubspecialtyFirm/list?search[filterid][subspecialty_id][value]=' . $subspecialtyId);
+		}
+
+		if (isset($search['filterid']['firm_id']['value']) && $search['filterid']['firm_id']['value'] > 0) {
+			$excludeSubspecialty = true;
+			$exludeFirm = false;
```
Comment (petergallagher) [inline protected/controllers/oeadmin/LeafletSubspecialtyFirmController.php]: There is a typo here, should be excludeFirm

Comment (vetusko) [inline protected/controllers/oeadmin/LeafletSubspecialtyFirmController.php]: fixed, pushed

**Comment by petergallagher** [comment]
Heading is OphTrConsent_Leaflet_Subspecialty but should be something human friendly.

When I try and add a leaflet I get "Property "OphTrConsent_Leaflet_Subspecialty.leaflet" is not defined.". The same thing happens with firm just with a different model name.

Also, the two drop downs at the top, from what I can see they are there to allow you to select a subspeciality or a firm to attach the leaflet to, but when I select a subspeciality firms disappear. Seeing as their is no subspeciality + firm mapping, only subspeciality or firm, should these not both always show all of the subspecialities and firms all of the time?

**Comment by vetusko** [comment]
- "Heading is OphTrConsent_Leaflet_Subspecialty but should be something human friendly."
Once it will be merged to develop it will display  "Leaflet-Subspecialty-Firm Assignment" - just changed the view in an other branch for that

- "only subspeciality or firm, should these not both always show all of the subspecialities and firms all of the time?"
No, once you select a subspecialty it will load data to the firm connected with that subspecialty only - that was a requirement

**Comment by petergallagher** [comment]
And can you also make it so when a form is submitted by GET the CSRF token is not sent in the URL.

**Pull request merged**